sqlparse/sql.py

```diff
@@ -163,6 +165,27 @@ def __init__(self, tokens=None):
         super().__init__(None, str(self))
         self.is_group = True
 
+    @property
+    def pos(self):
+        if len(self.tokens) > 0:
+            return self.tokens[0].pos
+
+    @property
+    def length(self):
+        if len(self.tokens) > 0:
+            first, last = self.tokens[0], self.tokens[-1]
+            if first.pos is not None and last.pos is not None:
+                return last.length + (last.pos - first.pos)
+
+        return len(str(self))
+
+    # this is a bit of a hack to avoid problems with the super constructor
+    # trying to set these attributes, which we want to compute dynamically
+    @pos.setter
+    def pos(self, value): ...
+    @length.setter
+    def length(self, value): ...
```

Comment:
I initially had the `Token` constructor do a `hasattr(self, 'pos')` check before assigning, which I didn't love either. But that approach broke one test where `deepcopy` was called on a parsed statement, whereas this approach is robust to that case.

It seemed work making these properties dynamic given that `TokenList` exposes methods for inserting tokens